### PR TITLE
Max Item Count Fix

### PR DESCRIPTION
**Pull request by bchong95**
# Max Item Count Fix

## Description

GROUP BY queries were not honoring `MaxItemCount`. None of the tests caught it. I updated the tests to assert that max item count is honored. The bug was a simple fix. 

## Type of change

Please delete options that are not relevant.

- [ x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Query/Core/ExecutionContext/PipelinedDocumentQueryExecutionContext.cs [hunk 1]
```diff
@@ -146,19 +146,18 @@ public override bool IsDone
...
+                // The query will block until all groupings are gathered so we might as well speed up the process.
                 initParams = new CosmosCrossPartitionQueryExecutionContext.CrossPartitionInitParams(
                     sqlQuerySpec: initParams.SqlQuerySpec,
                     collectionRid: initParams.CollectionRid,
                     partitionedQueryExecutionInfo: initParams.PartitionedQueryExecutionInfo,
                     partitionKeyRanges: initParams.PartitionKeyRanges,
-                    initialPageSize: initialPageSize,
-                    maxConcurrency: initParams.MaxConcurrency,
+                    initialPageSize: int.MaxValue,
+                    maxConcurrency: int.MaxValue,
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Query/Core/ExecutionContext/PipelinedDocumentQueryExecutionContext.cs]: What if the machine the application has limited resources and they don't want to consume so many threads?

File: Microsoft.Azure.Cosmos/src/Query/Core/ExecutionContext/PipelinedDocumentQueryExecutionContext.cs [hunk 2]
```diff
@@ -146,18 +146,17 @@ public override bool IsDone
 
             QueryInfo queryInfo = initParams.PartitionedQueryExecutionInfo.QueryInfo;
 
-            int actualPageSize = initParams.InitialPageSize;
             int initialPageSize = initParams.InitialPageSize;
             CosmosCrossPartitionQueryExecutionContext.CrossPartitionInitParams parameters = initParams;
             if (queryInfo.HasGroupBy)
             {
-                initialPageSize = int.MaxValue;
+                // The query will block until all groupings are gathered so we might as well speed up the process.
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Query/Core/ExecutionContext/PipelinedDocumentQueryExecutionContext.cs]: Please add a note to the changelog

Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Query/Core/ExecutionContext/PipelinedDocumentQueryExecutionContext.cs]: We never released the feature so the changelog doesn't need to be updated.

**Comment by j82w** [comment]
If max item count is currently broken please, and this fixes it please add it to the changelog.

File: Microsoft.Azure.Cosmos/src/Query/Core/ExecutionContext/PipelinedDocumentQueryExecutionContext.cs [hunk 3]
```diff
@@ -146,18 +146,17 @@ public override bool IsDone
 
             QueryInfo queryInfo = initParams.PartitionedQueryExecutionInfo.QueryInfo;
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Query/Core/ExecutionContext/PipelinedDocumentQueryExecutionContext.cs]: I'll approve it once the changelog is updated.

Comment (bchong95) [reply Microsoft.Azure.Cosmos/src/Query/Core/ExecutionContext/PipelinedDocumentQueryExecutionContext.cs]: I think we must be talking about different things. I have already stated that we never released the bug, so we don't need to update the changelog. If we have released the bug, then that is a different story and you just need to say so.

**Comment by bchong95** [comment]
@kirankumarkolli This is both a fix and code cleanup.

File: Microsoft.Azure.Cosmos/src/Query/Core/ExecutionContext/PipelinedDocumentQueryExecutionContext.cs (hunk 3 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos/src/Query/Core/ExecutionContext/PipelinedDocumentQueryExecutionContext.cs]: You are correct it's not in the 3.2.0 release. This needs to get ported to the 3.3.0 release.

**Pull request merged**